compiler/packages/react-compiler-runtime/src/index.ts

```diff
@@ -44,6 +44,41 @@ export function $read(memoCache: MemoCache, index: number) {
   return value;
 }
 
+export function u(
+  cache: any[],
+  getSource: () => any[],
+  conditionIndexs: number[],
+  conditionTargets: any[] | null,
+  updateIndexs: number[],
+  updateTargets: () => any[]
+) {
+  const test =
+    conditionTargets === null
+      ? (i: number) => cache[conditionIndexs[i]] === $empty
+      : (i: number) => cache[conditionIndexs[i]] !== conditionTargets[i];
+  const condition = (() => {
+    for (let i = 0; i < conditionIndexs.length; i++) {
+      if (test(i)) {
+        return true;
+      }
+    }
+    return false;
+  })();
```

Comment:
this is really inefficient. Let's turn this into static code without the unnecessary closure allocations. ie

```js
let hasChange = false;
for (let i = 0; i < conditionIndices.length; i++) {
   ...
   if (cache[conditionIndices[i]] !== conditionTargets[i]) {
     hasChange = true;
     break;
   }
   ...
}
```